### PR TITLE
Code Cleanup: Java 16 instanceof pattern matching

### PR DESCRIPTION
**Pull request by alshamams**
Pattern matching using instanceof operator in Java generally involves testing whether an object belongs to a certain type, and further converting the object to that type depending on the test result. 

From Java 16 onwards, this conversion step is avoided by using the type pattern as the second operand of instance operator itself. This helps make the code more concise.

This commit has tried to implement the above mentioned code change throughout.

Ref: https://github.com/eclipse-pde/eclipse.pde/issues/592

### CODE REVIEW

File: apitools/org.eclipse.pde.api.tools.annotations/.settings/org.eclipse.jdt.ui.prefs [hunk 1]
```diff
@@ -1,55 +1,145 @@
+cleanup.add_all=false
```
Comment (gireeshpunathil) [inline apitools/org.eclipse.pde.api.tools.annotations/.settings/org.eclipse.jdt.ui.prefs:1]: did you mean these changes (in preferences) to be made permanent?

File: ua/org.eclipse.pde.ua.ui/src/org/eclipse/pde/internal/ua/ui/editor/cheatsheet/comp/CompCSMasterTreeSection.java [hunk 2]
```diff
@@ -354,16 +353,9 @@ private void handlePreviewAction() {
...
+			if (input instanceof IFileEditorInput fileInput) {
 				url = fileInput.getFile().getLocationURI().toURL();
-			} else if (input instanceof IStorageEditorInput) {
-				// Note: This URL does not exist on the local file system
-				// As a result any tasks this composite cheat sheet has that
-				// specify a pathes to simple cheat sheets will not resolve
-				// Cheat sheet view will log an error loading simple cheat
-				// sheets
-				IStorageEditorInput storageInput = (IStorageEditorInput) input;
+			} else if (input instanceof IStorageEditorInput storageInput) {
```
Comment (akurtakov) [inline ua/org.eclipse.pde.ua.ui/src/org/eclipse/pde/internal/ua/ui/editor/cheatsheet/comp/CompCSMasterTreeSection.java:358]: Please keep the comment .

**Comment by iloveeclipse** [comment]
> This commit has tried to implement the above mentioned code change throughout.

Was it done manually or via some refactoring command?
Seeing 400 changed files makes me nervous, especially so close to M3.

**Comment by alshamams** [comment]
> > This commit has tried to implement the above mentioned code change throughout.
> 
> Was it done manually or via some refactoring command? Seeing 400 changed files makes me nervous, especially so close to M3.

Hi @iloveeclipse ,
This was done by using code cleanup from preferences, and this feature for pattern matching is readily available.

File: ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/PDEPlugin.java [hunk 3]
```diff
@@ -155,9 +155,7 @@ public static void logException(Throwable e, final String title, String message)
 			e = ((InvocationTargetException) e).getTargetException();
 		}
 		IStatus status = null;
-		if (e instanceof CoreException) {
-			// Re-use status only if it has attached exception with the stack trace
-			CoreException ce = (CoreException) e;
+		if (e instanceof CoreException ce) {
```
Comment (akurtakov) [inline ui/org.eclipse.pde.ui/src/org/eclipse/pde/internal/ui/PDEPlugin.java:158]: Please keep the comment.

File: ua/org.eclipse.pde.ua.ui/src/org/eclipse/pde/internal/ua/ui/editor/cheatsheet/comp/CompCSMasterTreeSection.java (hunk 2 above)
Comment (vik-chand) [reply ua/org.eclipse.pde.ua.ui/src/org/eclipse/pde/internal/ua/ui/editor/cheatsheet/comp/CompCSMasterTreeSection.java:358]: Is this a bug in the cleanup command?

**Comment by alshamams** [comment]
@iloveeclipse @akurtakov Henceforth I shall try to focus on making smaller commits. Should I revert and convert this into individual commits, for each project ?

File: ua/org.eclipse.pde.ua.ui/src/org/eclipse/pde/internal/ua/ui/editor/cheatsheet/comp/CompCSMasterTreeSection.java (hunk 2 above)
Comment (alshamams) [reply ua/org.eclipse.pde.ua.ui/src/org/eclipse/pde/internal/ua/ui/editor/cheatsheet/comp/CompCSMasterTreeSection.java:358]: > Is this a bug in the cleanup command?

Hi @vik-chand  ,
I did not remove the comment manually. It was done as part of the code cleanup.

**Comment by akurtakov** [comment]
Please do. That would allow us to deliver incrementally.

**Comment by alshamams** [comment]
Hi @iloveeclipse @akurtakov @vik-chand @gireeshpunathil 
I will be closing this PR, since I will be creating a new PR for each project, and dividing the entire work as you have suggested.

**Comment by vik-chand** [comment]
Sounds good ! Ideally we would want such incremental changes in M1/M2. M3 and RC1 are too close for comfort.

**Pull request closed without merging**